packages/components/date-picker/src/date-picker-com/panel-date-range.vue

```diff
@@ -655,18 +656,32 @@ const showTime = computed(
   () => props.type === 'datetime' || props.type === 'datetimerange'
 )
 
-const formatEmit = (emitDayjs: Dayjs | null, index?: number) => {
-  if (!emitDayjs) return
+const formatEmitWithEndOf = (
+  emitDayjs: Dayjs | null,
+  useEndOf: UnitTypeLong | UnitTypeShort | '' = ''
+) => {
+  return emitDayjs && useEndOf ? dayjs(emitDayjs).endOf(useEndOf) : emitDayjs
+}
+
+const formatEmit = (
+  emitDayjs: Dayjs | null,
+  index?: number,
+  useEndOf: UnitTypeLong | UnitTypeShort | '' = ''
```

Comment:
There is no need to pass parameter `useEndOf` here; the variables can be accessed directly from outside.